src/main/kotlin/app/revanced/patches/youtube/layout/theme/patch/ThemePatch.kt

```diff
@@ -45,9 +43,68 @@ class ThemePatch : ResourcePatch {
         }
 
         // copies the resource file to change the splash screen color
-        context.copyResources("theme",
-            ResourceUtils.ResourceGroup("values-night-v31", "styles.xml")
-        )
+        context.xmlEditor["res/values/attrs.xml"].use { editor ->
+            with(editor.file) {
+                val resourcesNode = getElementsByTagName("resources").item(0) as Element
+
+                val newElement: Element = createElement("attr")
+                newElement.setAttribute("format", "reference")
+                newElement.setAttribute("name", "splashScreenColor")
+
+                resourcesNode.appendChild(newElement)
+            }
+        }
+        context.xmlEditor["res/values/styles.xml"].use { editor ->
+            with(editor.file) {
+                val resourcesNode = getElementsByTagName("resources").item(0) as Element
+
+                for (i in 0 until resourcesNode.childNodes.length) {
+                    val node = resourcesNode.childNodes.item(i) as? Element ?: continue
+
+                    val newElement: Element = createElement("item")
+                    newElement.setAttribute("name", "splashScreenColor")
+
+                    when (node.getAttribute("name")) {
+                        "Base.Theme.YouTube.Launcher.Dark" -> {
+                            newElement.appendChild(createTextNode(darkThemeBackgroundColor))
+
+                            node.appendChild(newElement)
+                        }
+                        "Base.Theme.YouTube.Launcher.Light" -> {
+                            newElement.appendChild(createTextNode(lightThemeBackgroundColor));
+
+                            node.appendChild(newElement)
+                        }
+                    }
+                }
+            }
+        }
+        context.xmlEditor["res/values-v31/styles.xml"].use { editor ->
+            with(editor.file) {
+                val resourcesNode = getElementsByTagName("resources").item(0) as Element
+
+                val newElement: Element = createElement("item")
+                newElement.setAttribute("name", "android:windowSplashScreenBackground")
+
+                for (i in 0 until resourcesNode.childNodes.length) {
+                    val node = resourcesNode.childNodes.item(i) as? Element ?: continue
+
+                    if (node.getAttribute("name") == "Base.Theme.YouTube.Launcher") {
+                        newElement.appendChild(createTextNode("?attr/splashScreenColor"))
+
+                        node.appendChild(newElement)
+                    }
+                }
+            }
+        }
+        arrayOf("drawable", "drawable-sw600dp").forEach { drawablePath ->
```

Comment:
These are not all sizes, theres more dpis.

Reply:
> These are not all sizes, theres more dpis.

I checked better if there were other "quantum_launchscreen" files, but I didn't find any, except in the already listed folders.

Reply:
This is because the apk file you used does not have them, but depending on screen size you will get different different apks from play store.

Reply:
In this case I can't do anything else, because I have no way to access these files and decompile them to find the folder names.

Reply:
The folders exist:

![image](https://user-images.githubusercontent.com/13122796/209446813-5011ba83-4b00-450b-8bdc-cf78a72092a8.png)

Reply:
Ok so...I'll add all these to the list👍